### PR TITLE
Add support for multiple image.pullSecrets

### PR DESCRIPTION
**Pull request by mrombout**
At the moment it is only possible to set one pull secret through `image.pullSecret`. But for my use-case I would like to be able to use multiple `imagePullSecrets` to be able to pull the images from different private registries.

Some changes along the lines of this PR would be required, would a feature like this to acceptable to merge?

---

In this PR I went for the easy way and replaced `image.pullSecret` with `image.pullSecrets` with a bit of effort it's also possible to implement this in a way that retains compatibility.

### CODE REVIEW

**Comment by tobias-trabelsi-sonarsource** [comment]
Hi @mrombout ,

thanks for the contribution. can you clarify why you need to have multiple pull secrets? usually you would have a separated deployment if you have separated staging/production registries/cluster where you could just overwrite the pull secret. i currently can not really think of a reason to define multiple.

also this is not a blocker for the LTS, so changing the LTS chart is out of question for this, but we surely can improve for the current chart as well as for the DCE chart.

**Comment by mrombout** [comment]
None as of yet, but I intend on using this chart in a helmfile, so it's mainly the flexibility that I'm after. It's not too uncommon to have support for this (e.g. [influxdb](https://github.com/influxdata/helm-charts/tree/master/charts/influxdb), [postgresql](https://github.com/bitnami/charts/tree/master/bitnami/postgresql) (and other bitnami charts), etc.)

Having support multiple pull secrets would allow using `extraInitContainers` and `extraContainers` in environments that have the images for these containers on one private registry, and the DockerHub images on another private registry. And keep the secrets separate.

---

I've removed the changes to LTS and added the changes to DCE.

Since this is an incompatible change, should I change the major version. Or add back the old configuration parameters to keep compatibility and update the minor version?

**Comment by tobias-trabelsi-sonarsource** [comment]
where would something like this be used? i am not convinced to merge a breaking change only based on a "this might be useful some day"  gut feeling tbh. a private registry should be a trust anchor and having multiple registries that are accessible from the same cluster used for the same deployment does not improve security while just adding complexity.  

If you can give me a clear example about the concerns i am open to reconsider.

**Comment by larsla** [comment]
This would be useful for us since we want both authenticated access to Docker Hub (to work around the request quota) and being able to use an init container with an image from a private registry.

**Comment by tobias-trabelsi-sonarsource** [comment]
@larsla what you can already do. `extraInitContainers` needs to be passed as a [container spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#container-v1-core), so including a custom or additional pull secret

**Comment by tobias-trabelsi-sonarsource** [comment]
Okay so not using a private registry as a cache for the docker hub was not on my radar as a valid use case, but it makes sense to reduce storage requirements for the private registry. authenticating against the docker hub to get higher rate limits also is a valid use case.

@mrombout the DCE chart is still in beta, so `0.3.0` seems appropriate. for the SonarQube chart as this is a potential breaking change, a major version bump is needed :+1:

**Review by tobias-trabelsi-sonarsource — CHANGES_REQUESTED** [review]
Please mention your changes in the `CHANGELOG.md` as well as in the `Chart.yaml` so they are displayed on artifacthub.

also FYI: as this is a breaking change i pinged the responsible PM to make a final call on the versions. might take a little before we can merge

**Review by tobias-trabelsi-sonarsource — APPROVED** [review]
Functional review is okay.
Waiting for a decision from PM

**Comment by tobias-trabelsi-sonarsource** [comment]
Hi @mrombout 
happy new year. I had a discussion in between the years with my PM and he states that this improvement can not be applied with a breaking change. If you can provide a backwards compatible solution you can deprecate `image.pullSecret` in favor of `image.pullSecrets` and mention this. what do you think?

**Review by tobias-trabelsi-sonarsource — CHANGES_REQUESTED** [review]
Hi @mrombout 
happy new year. I had a discussion in between the years with my PM and he states that this improvement can not be applied with a breaking change. If you can provide a backwards compatible solution you can deprecate `image.pullSecret` in favor of `image.pullSecrets` and mention this. what do you think?

**Comment by tobias-trabelsi-sonarsource** [comment]
uhm is this really ready for review @mrombout ? this still seems like a breaking change to me. if a user did previously define `image.pullSecret`, this would no longer work after this change.

File: charts/sonarqube/Chart.yaml
Comment (tobias-trabelsi-sonarsource) [inline charts/sonarqube/Chart.yaml:4]: as this is no longer a breaking change a minor version bump should be enough

**Review by tobias-trabelsi-sonarsource — CHANGES_REQUESTED** [review]

File: charts/sonarqube/Chart.yaml
Comment (tobias-trabelsi-sonarsource) [reply charts/sonarqube/Chart.yaml:4]: same for the `CHANGELOG.md` else looks good :+1:

**Review by tobias-trabelsi-sonarsource — APPROVED** [review]
LGTM :+1: thanks for taking care of this :smiley:

**Pull request merged**
